llvm/tools/llvm-profdata/llvm-profdata.cpp

```diff
@@ -2695,30 +2695,30 @@ static void traverseAllValueSites(const InstrProfRecord &Func, uint32_t VK,
   uint32_t NS = Func.getNumValueSites(VK);
   Stats.TotalNumValueSites += NS;
   for (size_t I = 0; I < NS; ++I) {
-    uint32_t NV = Func.getNumValueDataForSite(VK, I);
-    std::unique_ptr<InstrProfValueData[]> VD = Func.getValueForSite(VK, I);
+    auto VD = Func.getValueArrayForSite(VK, I);
+    if (VD.empty())
```

Comment:
nit: instead of calling both empty and size, just call 'size()'. if NV is 0, return.

Reply:
Fixed in the latest iteration.